src/components/shoppingCart/cartdetail/CartDetail.jsx

```diff
@@ -3,11 +3,87 @@ import Container from "react-bootstrap/Container";
 import Row from "react-bootstrap/Row";
 import Col from "react-bootstrap/Col";
 import Button from "react-bootstrap/Button";
-import { MOCK_DATABASE } from "../../common/MockDatabase";
+import "./CartDetail.scss";
+import {
+  deleteItemFromCartAction,
+  modifyProductQuantity,
+  sumTotalPrice,
+  addOneMoreItem,
+  removeOneMoreItem,
+} from "../../redux/actions";
+import { useDispatch, useSelector } from "react-redux";
 
-const { title, images, brand, price } = MOCK_DATABASE[0];
+const CartDetail = (props) => {
+  const dispatch = useDispatch();
+  const productsData = useSelector((state) => state.addOrDeleteProductData);
+  // Getting all the total prices from individual items and add them together
+  const itemsTotalPricesArray = useSelector((state) => state.getTotalPrice);
+  const allItemsTotalPrice = itemsTotalPricesArray.reduce((cur, acc) => {
+    return cur + acc;
+  }, 0);
+
+  // Get total price from one item
+  const getItemTotalPrice = () => {
+    const sumTotal = props.info.price * props.info.quantity;
+    dispatch(sumTotalPrice(sumTotal));
+  };
+
+  React.useEffect(() => {
+    getItemTotalPrice();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // This function deletes thw whole product from the shopping cart and subtract the corresponding price from the total price
+  const deleteItemfromCartData = () => {
+    const copyOfData = productsData.slice();
+    for (let i = 0; i < copyOfData.length; i++) {
+      const currentItemId = props.info.id;
+      if (copyOfData[i].id === currentItemId) {
+        if (i > -1) {
+          copyOfData.splice(i, 1);
+        }
+        const singleItemTotalPrice = props.info.price * props.info.quantity;
+        const newTotal = allItemsTotalPrice - singleItemTotalPrice;
+        dispatch(removeOneMoreItem([newTotal]));
+        dispatch(deleteItemFromCartAction(copyOfData));
+        props.info.quantity = 1;
+        break;
+      }
+    }
+  };
+
+  const increaseAmountOfItems = () => {
+    const newTotal = allItemsTotalPrice + props.info.price;
+    dispatch(addOneMoreItem([newTotal]));
+  };
+
+  const decreaseAmountOfItems = () => {
+    const newTotal = allItemsTotalPrice - props.info.price;
```

Comment:
same i think all these computations are supposed to be in the reducers. the reducers are supposed to calculate this within redux so that the logic of the component is clean